java/source/flight.rst

```diff
@@ -0,0 +1,534 @@
+        public void getStream(CallContext context, Ticket ticket, ServerStreamListener listener) {
+            FlightDescriptor flightDescriptor = FlightDescriptor.path(
+                    new String(ticket.getBytes(), StandardCharsets.UTF_8));
+            Datasets datasets = this.datasets.get(flightDescriptor);
+            if (datasets == null) {
+                throw CallStatus.NOT_FOUND.withDescription("Unknown descriptor").toRuntimeException();
+            } else {
+                VectorSchemaRoot vectorSchemaRoot = VectorSchemaRoot.create(
+                        this.datasets.get(flightDescriptor).getSchema(), allocator);
+                listener.start(vectorSchemaRoot);
+                for (ArrowRecordBatch arrowRecordBatch : this.datasets.get(flightDescriptor).getbatches()) {
+                    VectorLoader loader = new VectorLoader(vectorSchemaRoot);
+                    loader.load(arrowRecordBatch.cloneWithTransfer(allocator));
+                    listener.putNext();
+                }
+                listener.completed();
+            }
+        }
+
+        @Override
+        public void doAction(CallContext context, Action action, StreamListener<Result> listener) {
+            FlightDescriptor flightDescriptor = FlightDescriptor.path(
+                    new String(action.getBody(), StandardCharsets.UTF_8));
+            switch (action.getType()) {
+                case "DELETE":
+                    if (datasets.remove(flightDescriptor) != null) {
+                        Result result = new Result("Delete completed".getBytes(StandardCharsets.UTF_8));
+                        listener.onNext(result);
+                    } else {
+                        Result result = new Result("Delete not completed. Reason: Key did not exist."
+                                .getBytes(StandardCharsets.UTF_8));
+                        listener.onNext(result);
+                    }
+                    listener.onCompleted();
+            }
+        }
+
+        @Override
+        public FlightInfo getFlightInfo(CallContext context, FlightDescriptor descriptor) {
+            FlightEndpoint flightEndpoint = new FlightEndpoint(
+                    new Ticket(descriptor.getPath().get(0).getBytes(StandardCharsets.UTF_8)), location);
+            return new FlightInfo(
+                    datasets.get(descriptor).getSchema(),
+                    descriptor,
+                    Collections.singletonList(flightEndpoint),
+                    /*bytes=*/-1,
+                    datasets.get(descriptor).getRows()
+            );
+        }
+
+        @Override
+        public void listFlights(CallContext context, Criteria criteria, StreamListener<FlightInfo> listener) {
+            datasets.forEach((k, v) -> { listener.onNext(getFlightInfo(null, k)); });
+            listener.onCompleted();
+        }
+    }
+    Location location = Location.forGrpcInsecure("0.0.0.0", 33333);
+    Schema schema = new Schema(Arrays.asList(
```

Comment:
nit, but create the schema closer to where it's actually used

Reply:
Updated